includes/Core/Authentication/Authentication.php

```diff
@@ -191,13 +211,26 @@ public function register() {
 		$this->verification_meta()->register();
 		$this->has_connected_admins->register();
 		$this->owner_id->register();
+		$this->connected_proxy_url->register();
+		$this->disconnected_reason->register();
 
-		add_action(
-			'init',
-			function() {
-				$this->handle_oauth();
-			}
-		);
+		$private_hook = function( $method ) {
+			return function() use ( $method ) {
+				return call_user_func_array( array( $this, $method ), func_get_args() );
+			};
+		};
```

Comment:
I'm usually a bit weary about abstractions around `add_action` and `add_filter` but since this is just a wrapper around _the callback itself_, it works for me!

However, since this pattern will be useful for many other classes too, we should extract this to a trait (in `Core\Util`) that any other class can use to get a method proxy function. This is untested but should work (also using PHP 5.6 features to avoid the need for `call_func_` functions):

```php
trait Method_Proxy_Trait {
	private function get_method_proxy( $method ) {
		return function ( ...$args ) use ( $method ) {
			return $this->{ $method }( ...$args );
		};
	}
}
```
Then in the class we can use it similarly:
```php
add_action( 'foo', $this->get_method_proxy( 'some_private_or_protected_method' ) );
```

While a little bit longer in usage (I don't like as much) it avoids the need for a temporary/local variable which might then be named/used inconsistently.

This could also be useful as a way to pass the method proxy function to another class as a way to expose a method as if it were public, but only to specific consumers.

Reply:
Added Method_Proxy_Trait trait.